### PR TITLE
[GPU] Fuse more 'type conversion only' reorders

### PR DESCRIPTION
**Pull request by yeonbok**
### Details:
 - Reorder which is only converting type can be fused to the prior node

### Tickets:
 - 144957

### CODE REVIEW

File: src/plugins/intel_gpu/tests/unit/fusions/eltwise_fusion_test.cpp [hunk 1]
```diff
@@ -545,21 +545,28 @@ INSTANTIATE_TEST_SUITE_P(fusings_gpu, eltwise_no_pitches_same_dims_quantize, ::t
...
-        input_layout("input2", get_input_layout2(p)),
-        eltwise("eltwise", { input_info("input"), input_info("input2") }, p.mode, p.default_type),
-        activation("activation", input_info("eltwise"), activation_func::relu, { 6.0f, 0.0f }),
-        reorder("out", input_info("activation"), p.default_format, data_types::f32)
-    );
+    create_topologies(input_layout("input", get_input_layout(p)),
+                      input_layout("input2", get_input_layout2(p)),
+                      eltwise("eltwise", {input_info("input"), input_info("input2")}, p.mode, p.default_type),
+                      activation("activation", input_info("eltwise"), activation_func::relu, {6.0f, 0.0f}),
+                      //        reorder("out", input_info("activation"), p.default_format, data_types::f32)
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/tests/unit/fusions/eltwise_fusion_test.cpp:555]: remove unnecessary comment block

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 2]
```diff
@@ -428,10 +430,18 @@ bool layout_optimizer::can_fuse_reorder_to_prev(program_node& prev, reorder_node
...
     // fusing reorder to the previous node can be done even if it is a dynamic shape case
-    if ((prev.is_type<mvn>() || prev.is_type<concatenation>()) &&
+    if ((prev.is_type<mvn>() || prev.is_type<concatenation>() || prev.is_type<gather>() || prev.is_type<broadcast>() ||
+         prev.is_type<select>() || prev.is_type<eltwise>()) &&
         !prev.is_in_shape_of_subgraph() && node.is_type_conversion_only() &&
-        (format::is_simple_data_format(fmt_prev) && format::is_simple_data_format(fmt_next)))
+        (format::is_simple_data_format(fmt_prev) && format::is_simple_data_format(fmt_next)) &&
+        // If the prev node is backedge of the loop, the type will be changed by fusing reorder.
+        // We can void only that case if we can check whether the current node is backedge of the network.
+        // However no such hanlde is existing yet. (TBD in the future when we need to optimize out the type converting
```
Comment (andrew-k-park) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:439]: nit: typo(hanlde -> handle)

**Review by andrew-k-park — APPROVED** [review]
LGTM except minor comments

File: src/plugins/intel_gpu/tests/unit/fusions/eltwise_fusion_test.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/unit/fusions/eltwise_fusion_test.cpp]: Fixed thanks!

**Pull request merged**